src/renderer/dx/DxRenderer.cpp

```diff
@@ -1484,63 +1484,84 @@ try
 
     _d2dBrushForeground->SetColor(_ColorFFromColorRef(color));
 
-    const auto font = _glyphCell;
-    D2D_POINT_2F target = til::point{ coordTarget } * font;
+    const D2D1_SIZE_F font = _glyphCell;
+    const D2D_POINT_2F target = { coordTarget.X * font.width, coordTarget.Y * font.height };
+    const auto fullRunWidth = font.width * gsl::narrow_cast<unsigned>(cchLine);
 
-    D2D_POINT_2F start = { 0 };
-    D2D_POINT_2F end = { 0 };
-
-    for (size_t i = 0; i < cchLine; i++)
+    const auto DrawLine = [=](const auto x0, const auto y0, const auto x1, const auto y1, const auto strokeWidth) noexcept
     {
-        // 0.5 pixel offset for crisp lines
-        start = { target.x + 0.5f, target.y + 0.5f };
+        _d2dDeviceContext->DrawLine({ x0, y0 }, { x1, y1 }, _d2dBrushForeground.Get(), strokeWidth, _strokeStyle.Get());

```

Comment:
By using [=] for this lambda, are we having an AddRef/Release occur to the underlying elements when the ComPtrs are copied for use inside this lambda? Perhaps [&] would be better in this case?

Reply:
I KNEW IT. I knew that I thought I finally found something in a James PR and it turns out that I'm the one who is wrong. Pffsh. Figures.

@DHowett assures me he thought the same thing here but it's really `this` that is copied byval into the lambda and since it's used only in the context of this same function, everything is just peachy.

Reply:
I wasn't really sure which option was best, but I looked at the generated assembly and it made no difference either way - in both case the lambda inlined to the exact same code.